quotefault/__init__.py

```diff
@@ -215,28 +239,138 @@ def get():
 @app.route('/additional', methods=['GET'])
 @auth.oidc_auth
 def additional_quotes():
+    """
+    Show beyond the first 20 quotes
+    """
 
     metadata = get_metadata()
-    metadata['submitter'] = request.args.get('submitter')  # get submitter from url query string
-    metadata['speaker'] = request.args.get('speaker')  # get speaker from url query string
-
-    #return quotes according to query strings (or lack thereof)
-    if metadata['speaker'] is not None and metadata['submitter'] is not None:
-        quotes = db.session.query(Quote, func.sum(Vote.direction).label('votes')).outerjoin(Vote).group_by(Quote).order_by(Quote.quote_time.desc()).filter(Quote.submitter == metadata['submitter'], Quote.speaker == metadata['speaker']).all()
-    elif metadata['submitter'] is not None:
-        quotes = db.session.query(Quote, func.sum(Vote.direction).label('votes')).outerjoin(Vote).group_by(Quote).order_by(Quote.quote_time.desc()).filter(Quote.submitter == metadata['submitter']).all()
-    elif metadata['speaker'] is not None:
-        quotes = db.session.query(Quote, func.sum(Vote.direction).label('votes')).outerjoin(Vote).group_by(Quote).order_by(Quote.quote_time.desc()).filter(Quote.speaker == metadata['speaker']).all()
-    else:
-        quotes = db.session.query(Quote, func.sum(Vote.direction).label('votes')).outerjoin(Vote).group_by(Quote).order_by(Quote.quote_time.desc()).all()
+
+    # Get all the quotes
+    quotes = get_quote_query(speaker = request.args.get('speaker'),
+        submitter = request.args.get('submitter')).all()
 
     #tie any votes the user has made to their uid
     user_votes = db.session.query(Vote).filter(Vote.voter == metadata['uid']).all()
 
-
     return render_template(
         'bootstrap/additional_quotes.html',
         quotes=quotes[20:],
         metadata=metadata,
         user_votes=user_votes
     )
+
+@app.route('/report/<report_id>', methods=['POST'])
+@auth.oidc_auth
+def submit_report(report_id):
+    """
+    Report a quote and notify EBoard/RTP
+    """
+    metadata = get_metadata()
+    existing_report = Report.query.filter(Report.reporter==metadata['uid'],
+        Report.quote_id==report_id).first()
+    if existing_report:
+        flash("You already submitted a report for this Quote!")
+        return redirect('/storage')
+    new_report = Report(report_id, metadata['uid'], None)
+    db.session.add(new_report)
+    db.session.commit()
+    if app.config['MAIL_SERVER'] != '':
+        send_report_email( metadata['uid'], Quote.query.get(report_id) )
+
+    flash("Report Successful!")
+    return redirect('/storage')

```

Comment:
If the user loaded the page with query params, will this redirect lose them?

Reply:
Yes

Reply:
😿 is there a way we could make this not annoying for users? Page reloads aren't the nicest, specially when you might have to scroll a lot to get to where you were.